cmd/soroban-rpc/internal/db/db.go

```diff
@@ -15,7 +15,9 @@ import (
 
 	"github.com/stellar/go/support/db"
 	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/support/log"
 	"github.com/stellar/go/xdr"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/daemon/interfaces"
```

Comment:
does this cause a circular import? :/

Reply:
@Shaptic Nope, I am able to compile this branch and run RPC successfully.

Reply:
me too, but according to https://github.com/golang/go/issues/51180 it may cause issues in tests?? even though mine pass just fine locally

Reply:
Even if it works locally, I think we should not import `daemon/interface` here because db.go is at more lower level which should not have access to daemon package. Currently what we need is only `MetricsNamespace` and `MetricsRegistry` in db.go. 

It is circular bc daemon is already using db.go, it might be better to refactor now imo.

Reply:
I think it's technically not circular because `daemon` imports `db` but `db` imports `daemon/interface`? So they're different. I totally agree that daemon has no business here but this came from review comments (https://github.com/stellar/soroban-rpc/pull/141#discussion_r1579522946) on my PR.